packages/superset-ui-query/src/api/v1/postChartData.ts

```diff
@@ -7,17 +7,15 @@ export interface Params extends BaseParams {
   queryContext: QueryContext;
 }
 
-export default function postChartData({
+export default async function postChartData({
```

Comment:
This is broader js question. 
What is the advantage of `async` vs returning promise? I sometimes not sure what do I get from converting to async function, which sometimes ask for the `regenerator-runtime` when transpile.

Reply:
`async/await` reduces parentheses/indent levels and makes the code somewhat more straightforward IMO.

The missing `regenerator-runtime` error is annoying but there is a known solution for that so I can live with it.

Reply:
Let's not do it in this PR yet. 
To adopt this, better make all the api wrapper functions `/api/**/*.ts` adopt it for consistency.

Reply:
I changed all of those in #635 . I'd rather not have to revert these changes as there is already a mix of both in use.

Reply:
ok